liblangutil/Scanner.cpp

```diff
@@ -764,7 +764,7 @@ bool Scanner::scanEscape()
 	char c = m_char;
 
 	// Skip escaped newlines.
```

Comment:
Also, one way to make this code more self-explanatory would be to add a meaningfuly named parameter (e.g. `_preserveInvalidEscapes` or `_rejectInvalidEscapes` with the opposite meaning) instead of special-casing the `SpecialComment` mode here. It would make it clearer here why we're doing something different in this mode.

Then, at the point of call that would also make it more obvious that the function will behave differently:

```c++
		if (m_kind == ScannerKind::SpecialComment)
		{
			if (c == '\\')
				scanEscape(true /* _preserveInvalidEscapes */);
```

Reply:
True, I have opted for `_rejectInvalidEscapes` since then it's no negations thereof in the implementation (or I would've had to reorganize the if statements).